File: gherkin/dotnet/Gherkin/Events/Args/Location.cs
using System.Runtime.Serialization;

namespace Gherkin.Events.Args
{
    public class Location
    {
        public Location(int column, int line)
        {
            Column = column == 0 ? (int?)null : column;
            Line = line;
        }

        public Location()
        {
        }

        [DataMember(Name = "column")]
        public int? Column { get; set; }
        [DataMember(Name = "line")]
        public int Line { get; set; }
    }
}

File: gherkin/php/src/Location.php
<?php

declare(strict_types=1);

namespace Cucumber\Gherkin;

final class Location
{
    public function __construct(
        public readonly int $line,
        public readonly int $column,
    ) {
    }

    public function getColumn(): int
    {
        return $this->column;
    }
}
